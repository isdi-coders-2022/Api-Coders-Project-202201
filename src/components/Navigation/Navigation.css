.navigation__list {
  margin: 0;
  padding: 0;

  display: flex;
  justify-content: space-around;
  align-items: center;

  position: fixed;
  bottom: 0;

  width: 100vw;
  height: 60px;

  border-top: 1px solid #eaeaea;
  background-color: white;
  color: black;

  box-shadow: 0px 0px 20px 5px #eaeaea;
  z-index: 100;
}

a {
  color: black;
}

a:visited {
  color: black;
}

.navigation__text {
  display: none;
}

.navigation__list li {
  list-style: none;
}

.navigation__icon {
  font-size: 30px;
}

.add {
  position: absolute;
  bottom: 30px;

  width: 60px;
  height: 60px;
  background-color: white;
  border: 2px solid #eaeaea;
  border-radius: 50%;

  display: flex;
  justify-content: center;
  align-items: center;

  box-shadow: 0px -2px 10px 1px #eaeaea;
}
