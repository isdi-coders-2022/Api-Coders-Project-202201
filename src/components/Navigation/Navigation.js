import "./Navigation.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faGlassMartiniAlt,
  faHouse,
  faPlus,
} from "@fortawesome/free-solid-svg-icons";

const Navigation = () => {
  return (
    <nav className="navigation">
      <ul className="navigation__list">
        <li className="navigation__item home">
          <FontAwesomeIcon icon={faHouse} className="navigation__icon" />
          <span className="navigation__text">Home</span>
        </li>

        <li className="navigation__item glass">
          <FontAwesomeIcon
            icon={faGlassMartiniAlt}
            className="navigation__icon"
          />
          <span className="navigation__text">My Bar</span>
        </li>

        <li className="navigation__item add">
          <FontAwesomeIcon
            icon={faPlus}
            className="navigation__icon plusIcon"
          />
          <span className="navigation__text">Create My Cocktail</span>
        </li>
      </ul>
    </nav>
  );
};

export default Navigation;
