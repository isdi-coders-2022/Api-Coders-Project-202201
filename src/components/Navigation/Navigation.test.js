import { render, screen } from "@testing-library/react";
import Navigation from "./Navigation";

describe("Given a Navigation component", () => {
  describe("When it's rendered", () => {
    test("Then it should display a link with 'Home' as text", () => {
      render(<Navigation />);

      const navigation = screen.queryByRole("link", { name: "Home" });

      expect(navigation).toBeInTheDocument();
    });
  });
});
