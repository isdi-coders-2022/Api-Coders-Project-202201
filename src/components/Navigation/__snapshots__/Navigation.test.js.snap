// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Navigation component When Then 1`] = `
<nav
  className="navigation"
>
  <ul
    className="navigation__list"
  >
    <li
      className="navigation__item home"
    >
      <a
        href="home"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-house fa-w-18 navigation__icon"
          data-icon="house"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M575.8 255.5C575.8 273.5 560.8 287.6 543.8 287.6H511.8L512.5 447.7C512.5 450.5 512.3 453.1 512 455.8V472C512 494.1 494.1 512 472 512H456C454.9 512 453.8 511.1 452.7 511.9C451.3 511.1 449.9 512 448.5 512H392C369.9 512 352 494.1 352 472V384C352 366.3 337.7 352 320 352H256C238.3 352 224 366.3 224 384V472C224 494.1 206.1 512 184 512H128.1C126.6 512 125.1 511.9 123.6 511.8C122.4 511.9 121.2 512 120 512H104C81.91 512 64 494.1 64 472V360C64 359.1 64.03 358.1 64.09 357.2V287.6H32.05C14.02 287.6 0 273.5 0 255.5C0 246.5 3.004 238.5 10.01 231.5L266.4 8.016C273.4 1.002 281.4 0 288.4 0C295.4 0 303.4 2.004 309.5 7.014L564.8 231.5C572.8 238.5 576.9 246.5 575.8 255.5L575.8 255.5z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <span
          className="navigation__text"
        >
          Home
        </span>
      </a>
    </li>
    <li
      className="navigation__item glass"
    >
      <a
        href="mybar"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-martini-glass fa-w-16 navigation__icon"
          data-icon="martini-glass"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502 57.63C523.3 36.38 508.3 0 478.3 0H33.72C3.711 0-11.29 36.38 9.962 57.63l214 214V448H175.1c-26.51 0-47.1 21.49-47.1 48c0 8.836 7.164 16 16 16h224c8.836 0 16-7.164 16-16c0-26.51-21.49-48-48-48h-47.1V271.6L502 57.63zM405.1 64l-64.01 64H170.9L106.9 64H405.1z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <span
          className="navigation__text"
        >
          My Bar
        </span>
      </a>
    </li>
    <li
      className="navigation__item add"
    >
      <a
        href="form"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-plus fa-w-14 navigation__icon plusIcon"
          data-icon="plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M432 256c0 17.69-14.33 32.01-32 32.01H256v144c0 17.69-14.33 31.99-32 31.99s-32-14.3-32-31.99v-144H48c-17.67 0-32-14.32-32-32.01s14.33-31.99 32-31.99H192v-144c0-17.69 14.33-32.01 32-32.01s32 14.32 32 32.01v144h144C417.7 224 432 238.3 432 256z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <span
          className="navigation__text"
        >
          Create My Cocktail
        </span>
      </a>
    </li>
  </ul>
</nav>
`;
